### PR TITLE
[Android] Fix ImageButton Padding the third

### LINKED ISSUE #13101: Padding on ImageButtons works wrong on Android
### Description

If you add padding to an ImageButton on Android, not only a padding will be set, but also a margin with the size of the padding. The button will therefore become smaller due to the padding. To get the button back to the right size, you then need to set a margin with a negative size.

### Steps to Reproduce

1. Create a new Maui App
2. Create a Grid
3. Add a ImageButton to the Grid
4. Set a Padding on the ImageButton
5. Add a Button to the Grid to see the Size difference
6. It should be now like:
```xaml
 <Grid ColumnDefinitions="*,100,100,*" RowDefinitions="*,100,*">
        <ImageButton Grid.Column="1" Grid.Row="1" Padding="10" BackgroundColor="Aqua" Source="dotnet_bot.png"/>
        <Button Grid.Column="2" Grid.Row="1" Text="testButton" Padding="10" BackgroundColor="Blue" WidthRequest="100" HeightRequest="100"/>
    </Grid>
```
### Link to public reproduction project repository

https://github.com/matzeteo/TestApps.git

### Version with bug

8.0 (current)

### Affected platforms

Android

### Affected platform versions

Android 10.0 and up

### Did you find any workaround?

Set a margin with the negative value of the padding. But that doesn't work for every Layout structure.

### Relevant log output

_No response_

### LINKED ISSUE #16713: ImageButton Padding not working when manipulate IsVisible On Android
### Description

If IsVisible of the ImageButton or IsVisible of another View on the same level as the ImageButton changes, the Padding of the ImageButton is disabled.

```xml
<AbsoluteLayout>        

    <ImageButton
        AbsoluteLayout.LayoutFlags="PositionProportional"
        AbsoluteLayout.LayoutBounds="0.5,0.5,120,120"
        Source="dotnet_bot.png"
        WidthRequest="120"
        HeightRequest="120"
        CornerRadius="60"
        Padding="30"
        Aspect="AspectFit"                
        BackgroundColor="Beige"
        IsVisible="true"
        x:Name="imageButton"/>

    <ContentView
        AbsoluteLayout.LayoutFlags="SizeProportional"
        AbsoluteLayout.LayoutBounds="0,0,1.0,1.0"
        BackgroundColor="#80FFFFFF"
        IsVisible="true"
        x:Name="overlay"
        />
        
</AbsoluteLayout>
```

```cs
    public MainPage()
    {
        InitializeComponent();

        Dispatcher.DispatchDelayed(TimeSpan.FromSeconds(3), () =>
        {
            overlay.IsVisible = false;
            // workaround
            //this.ForceLayout();
        });
    }
```

| Before | After |
|:-:|:-:|
|||

### Steps to Reproduce

1. Arrenge an ImageView and any other View on  the same parent layout .
2. Manipulate IsVisibleProperty of either.
3. ImageView Padding disabled.

### Link to public reproduction project repository

https://github.com/muak/Maui.ImageButtonIssue

### Version with bug

8.0.0-preview.7.8842

### Is this a regression from previous behavior?

Yes, this used to work in Xamarin.Forms, Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android13

### Did you find any workaround?

This can be avoided by calling Page's ``ForceLayout``.

### Relevant log output

_No response_

### LINKED ISSUE #18001: [ImageButton][Android] Padding is incorrect after IsVisible is changed
### Description

Hello, and thank you for this awesome framework.

I've noticed a bug where ImageButtons for Android does not correctly set their Padding. I first thought it happened every time you draw a ImageButton, but that was not the case. It happens whenever you change the IsVisible property, which in many cases people are I presume.

In the follow example I have a `ImageButton` with the `dotnet_bot` as source and a red background. I've set the Padding to `50`, which should make the bot smaller inside the button. This works the first time the app loads, but once I change the IsVisible (using the `Switch`) it ignores the Padding.

https://github.com/dotnet/maui/assets/2527084/68a41a7f-40f5-4b0a-a391-6e94ede95597

### Steps to Reproduce

1. [Clone this repository.](https://github.com/haavamoa/MauiApp/tree/task/hmo-imagebutton-padding-android)
2. Select Android
3. Run the code
4. Observe the `ImageButton` has the correct padding once it loads.
5. Observe the `ImageButton` not having the correct padding when you tap the `Switch`. 

### Link to public reproduction project repository

https://github.com/haavamoa/MauiApp/tree/task/hmo-imagebutton-padding-android

### Version with bug

8.0.0-rc.2.9373

### Is this a regression from previous behavior?

Yes, this used to work in Xamarin.Forms

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

_No response_

### Did you find any workaround?

The way to get around this the issue where IsVisible is flipped is to make sure to update the padding after the IsVisible has changed:

[Here is our fix, where we ship a wrapping ImageButton class.](https://github.com/DIPSAS/DIPS.Mobile.UI/blob/cb4d8c854656911561f37071f41af76c6647fa8c/src/library/DIPS.Mobile.UI/Components/Images/ImageButton/ImageButton.cs#L7)

### Relevant log output

_No response_

### LINKED ISSUE #25201: [Android] Invisible ImageButton Padding Broken
### Description

On Android setting a Padding on a not visible ImageButton produces incorrect results when it becomes visible.
Expected
Image

Actual
Image

### Steps to Reproduce

Clone repo, start and click on button "click me". Observe incorrect ImageButton.
or
1. Add an ImageButton like so:
```xaml
<ImageButton
    Background="Red"
    Padding="50"
    IsVisible="False"

    Source="dotnet_bot.png"
    HeightRequest="185"
    x:Name="imageButton"
    />
```
2. Make the ImageButton visible in a button click event (`imageButton.IsVisible = true;`).

### Link to public reproduction project repository

https://github.com/aheubusch/MauiAndroidImageButtonPadding

### Version with bug

8.0.91 SR9.1

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

_No response_

### Affected platforms

Android

### Affected platform versions

_No response_

### Did you find any workaround?

Nope :(

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by aheubusch**
### Description of Change

Improve the workaround (#22298) for the iffy Android ShapeImageView contentPadding/padding handling.
The underlying problem is that ShapeImageView updates Padding in onMeasure leading to a double padding. The previous workaround reset the Padding after one eventloop hoping that onMeasure was called in the meantime. This worked for most cases but not all. The "improved" workaround overrides onMeasure and resets the Padding as soon as it was set.

I ran all ImageButton tests I could find and they are all green. So thats a good sign.

### Issues Fixed

Fixes #25201 
Fixes #16713
Fixes #18001
Fixes #13101

### CODE REVIEW

File: src/Core/src/Platform/Android/ImageButtonExtensions.cs [hunk 1]
```diff
@@ -29,29 +29,9 @@ public static async void UpdatePadding(this ShapeableImageView platformButton, I
...
-			await Task.Yield();
-
-			if (!platformButton.IsAlive())
-				return;
-
-			// We must re-set all the paddings because the first time was not hard enough.
-			platformButton.SetContentPadding((int)padding.Left, (int)padding.Top, (int)padding.Right, (int)padding.Bottom);
-			platformButton.SetPadding(0, 0, 0, 0);
-
-			// Just like before, the bugs are not done. This needs to trigger a re-calculation of
```
Comment (aheubusch) [inline src/Core/src/Platform/Android/ImageButtonExtensions.cs:46]: @mattleibow You have added this part (ShapeAppearanceModel) in #22298. I am not sure if I can just remove it. None of the tests that looked relevant failed on my machine.

**Comment by aheubusch** [comment]
Not sure if the test failure are because of this PR. They all fail with a `System.TimeoutException`.

File: src/Core/src/Platform/Android/ImageButtonExtensions.cs (hunk 1 above)
Comment (mattleibow) [reply src/Core/src/Platform/Android/ImageButtonExtensions.cs:49]: I am scared tbh. I forget the corner cases where this happens - and they yield is especially scary. I think I was going off the fixes the native side was doing.

The current test set may not be enough and we will have to go remove this and see what breaks manually, and then make your change to see if it fixes it.

**Review by mattleibow — COMMENTED** [review]
I am super scared with this as the ImageButton was super fragile and I think I have seen padding issues but was super hard to repro in a test or I was not able to reliably repro.

I added tests as I did things, and TJ probably added more. But, I am not sure there are enough.

This comment does not help move the PR along, but I need to play a bit more and try and repro the issues that required the terrible hacks.

**Comment by BlueRaja** [comment]
Any chance to get this in for the next release? Both `Border` and `Padding` on `ImageButton` are [still broken](https://github.com/dotnet/maui/issues/13101#issuecomment-2645984846), with no known workaround. This seems like pretty basic functionality to just straight up not work at all.

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue25201.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
+﻿#if ANDROID
```
Comment (jsuarezruiz) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue25201.cs:1]: Could we run the test in all the platforms? In that way, we can just verify that the Padding property behavior is aligned.

**Comment by karthikraja-arumugam** [comment]
This PR also resolves the issue https://github.com/dotnet/maui/issues/18875

**Review by rmarinho — APPROVED** [review]

File: src/Core/src/Platform/Android/MauiShapeableImageView.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
+﻿using Android.Content;
+using Android.Runtime;
+using Android.Util;
+using Google.Android.Material.ImageView;
+
+namespace Microsoft.Maui.Platform
+{
+	internal class MauiShapeableImageView : ShapeableImageView
```
Comment (mattleibow) [inline src/Core/src/Platform/Android/MauiShapeableImageView.cs:8]: Just looking at this and I think that this is all Java code so we probably should just do it in Java and let the bonding do the magic.

Doing it in C# is probably going have a big perf impact.

**Review by mattleibow — CHANGES_REQUESTED** [review]
Now that I have had a nap, I don't think we should be writing C# anymore ;) OK, maybe we just need to put that new maui shape view into Java and use it there. The hopping back and forth will be a perf hit.

**Comment by mattleibow** [comment]
We can do Java in a new PR: https://github.com/dotnet/maui/issues/28532

File: src/Core/src/Platform/Android/MauiShapeableImageView.cs (hunk 3 above)
Comment (mattleibow) [reply src/Core/src/Platform/Android/MauiShapeableImageView.cs:8]: New issue for this Java work: https://github.com/dotnet/maui/issues/28532

**Review by mattleibow — APPROVED** [review]
